feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/signer/proxy/ProxiedSigner.kt

```diff
@@ -73,21 +73,30 @@ class ProxiedSigner(
         return delegate.signerAccountId(chain)
     }
 
+    override suspend fun modifyPayload(payloadExtrinsic: SignerPayloadExtrinsic): SignerPayloadExtrinsic {
+        val chain = chainRegistry.getChain(payloadExtrinsic.chainId)
+        val proxyMetaAccount = getProxyMetaAccount()
+        val delegate = createDelegate(proxyMetaAccount)
+        val payload = checkPermissionAndWrap(proxyMetaAccount, payloadExtrinsic, chain)
+        return delegate.modifyPayload(payload)
+    }
+
     override suspend fun signExtrinsic(payloadExtrinsic: SignerPayloadExtrinsic): SignedExtrinsic {
         val chain = chainRegistry.getChain(payloadExtrinsic.chainId)
         val proxyMetaAccount = getProxyMetaAccount()
 
-        acknowledgeProxyOperation(proxyMetaAccount)
+        if (isRootProxied) {
+            acknowledgeProxyOperation(proxyMetaAccount)
+        }
+
+        val payloadToSign = if (isRootProxied) modifyPayload(payloadExtrinsic) else payloadExtrinsic
 
         if (isRootProxied) {
-            validateExtrinsic(payloadExtrinsic, chain)
+            validateExtrinsic(payloadToSign, chain)
         }
 
         val delegate = createDelegate(proxyMetaAccount)
-        val modifiedPayload = modifyPayload(proxyMetaAccount, payloadExtrinsic, chain)
-
-        val signedExtrinsic = delegate.signExtrinsic(modifiedPayload)
-        return signedExtrinsic
+        return delegate.signExtrinsic(payloadToSign)
```

Comment:
I dont quite like the idea of letting `signExtrinsic` handle both final and non-final payloads. It might quickly become error pronse as we add more signer types like multisig. `modifyPayload` can return both "one layer deep" wrapped payload and "final payload" so `signExtrinsic` might use final payload for validation and "one layer deep" to pass it to the bottom layer.